kernel/common/enumerable.rb

```diff
@@ -212,6 +212,12 @@ def slice_when(&block)
     end
   end
 
+  def chunk_while(&block)
+    raise ArgumentError, "no block given" unless block_given?
+
+    slice_when { |before, after| !(block.call(before, after)) }
```

Comment:
The explicit block isn't needed here, this can just be:

    !(yield before, after)